src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -240,9 +238,13 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private Activity StartActivity(HttpContext httpContext, StringValues requestId)
+        private Activity StartActivity(HttpContext httpContext)
         {
             var activity = new Activity(ActivityName);
+
+            if (!httpContext.Request.Headers.TryGetValue(TraceParentHeaderName, out var requestId))
```

Comment:
Could you please also read the tracestate if traceparent is present?

Reply:
Done